### PR TITLE
Configure initial tablets count scaling

### PR DESCRIPTION
**Pull request by xemul**
There are currently two options how to "request" the number of initial tables for a table

1. specify it explicitly when creating a keyspace
2. let scylla calculate it on its own

Both are not very nice. The former doesn't take cluster layout into consideration. The latter does, but starts with one tablet per shard, which can be too low if the amount of data grows rapidly.

Here's a (maybe temporary) proposal to facilitate at least perf tests -- the --tablets-initial-scale-factor option that enhances the option number two above by multiplying the calculated number of tablets by the configured number. This is what we currently do to run perf tests by patching scylla, with the option it going to be more convenient.

### CODE REVIEW

**Comment by tgrabiec** [comment]
Such parameters are really global, so long term we probably should keep them in group0 like schema. They could be populated initially from yaml, when the first node is bootstrapped. Later you would change it via CQL, where mutations to system.global_config are applied via group0.

**Review by tgrabiec — APPROVED** [review]

File: service/tablet_allocator.cc [hunk 1]
```diff
@@ -804,7 +804,9 @@ class tablet_allocator_impl : public tablet_allocator::impl
             : _config(std::move(cfg))
             , _migration_notifier(mn)
             , _db(db) {
-        (void)_config;
+        if (_config.initial_tablets_scale == 0) {
+            throw std::runtime_error("Initial tablets scale must be positive");
```
Comment (denesb) [inline service/tablet_allocator.cc:808]: nit: 0 is positive, the error should say "positive, non-zero".

Comment (xemul) [reply service/tablet_allocator.cc:808]: https://www.mathopenref.com/positive-number.html:
> Positive [numbers](https://www.mathopenref.com/number.html) are numbers that are greater than zero.

Comment (denesb) [reply service/tablet_allocator.cc:808]: :laughing: 

Well played, queued.

**Pull request merged**